src/Symfony/Component/Console/Output/ConsoleOutput.php

```diff
@@ -125,7 +125,13 @@ protected function hasStderrSupport()
      */
     private function isRunningOS400()
     {
-        return 'OS400' === PHP_OS;
+        $checks = array(
+            function_exists('php_uname') ? php_uname('s') : '',
+            getenv('OSTYPE'),
+            defined('PHP_OS') ? PHP_OS : '',
```

Comment:
This check can be omitted IMO, we can be sure that the  `PHP_OS` constant exists.